src/Sulu/Bundle/MediaBundle/Media/FormatOptions/FormatOptionsManager.php

```diff
@@ -0,0 +1,255 @@
+     * @param MediaManagerInterface $mediaManager
+     * @param FormatManagerInterface $formatManager
+     * @param array $formats
+     */
+    public function __construct(
+        EntityManagerInterface $em,
+        EntityRepository $formatOptionsRepository,
+        MediaManagerInterface $mediaManager,
+        FormatManagerInterface $formatManager,
+        array $formats
+    ) {
+        $this->em = $em;
+        $this->formatOptionsRepository = $formatOptionsRepository;
+        $this->mediaManager = $mediaManager;
+        $this->formatManager = $formatManager;
+        $this->formats = $formats;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($mediaId, $formatKey)
+    {
+        if (!isset($this->formats[$formatKey])) {
+            throw new FormatNotFoundException($formatKey);
+        }
+
+        $media = $this->mediaManager->getEntityById($mediaId);
+        $fileVersion = $this->getFileVersionForMedia($media);
+
+        /** @var FormatOptions $formatOptions */
+        $formatOptions = $this->formatOptionsRepository->find(
+            [
+                'fileVersion' => $fileVersion,
+                'formatKey' => $formatKey,
+            ]
+        );
+
+        if (!isset($formatOptions)) {
+            return [];
+        }
+
+        return $this->entityToArray($formatOptions);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAll($mediaId)
+    {
+        $media = $this->mediaManager->getEntityById($mediaId);
+        $fileVersion = $this->getFileVersionForMedia($media);
+
+        $formatOptions = $this->formatOptionsRepository->findBy(['fileVersion' => $fileVersion]);
+
+        $formatOptionsArray = [];
+
+        /** @var FormatOptions $formatOptionEntity */
+        foreach ($formatOptions as $formatOptionEntity) {
+            $formatOptionsArray[$formatOptionEntity->getFormatKey()] = $this->entityToArray($formatOptionEntity);
+        }
+
+        return $formatOptionsArray;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save($mediaId, $formatKey, array $data)
+    {
+        if (!isset($this->formats[$formatKey])) {
+            throw new FormatNotFoundException($formatKey);
+        }
+
+        $media = $this->mediaManager->getEntityById($mediaId);
+        $fileVersion = $this->getFileVersionForMedia($media);
+
+        $formatOptions = $fileVersion->getFormatOptions()->get($formatKey);
+        if (!isset($formatOptions)) {
+            $formatOptions = new FormatOptions();
+            $formatOptions->setFileVersion($fileVersion);
+            $fileVersion->addFormatOptions($formatOptions);
+            $formatOptions->setFormatKey($formatKey);
+        }
+
+        $formatOptions = $this->setDataOnEntity($formatOptions, $data);
+        $fileVersion->setSubVersion($fileVersion->getSubVersion() + 1);
+
+        $this->em->persist($formatOptions);
+        $this->em->persist($fileVersion);
+        $this->em->flush();
```

Comment:
We actually decided not to flush in managers, this should be done in the controller or commands.